src/github.com/matrix-org/dendrite/syncapi/storage/syncserver.go

```diff
@@ -209,7 +215,11 @@ func (d *SyncServerDatabase) syncStreamPositionTx(
 	return types.StreamPosition(maxID), nil
 }
 
-// IncrementalSync returns all the data needed in order to create an incremental sync response.
+// IncrementalSync returns all the data needed in order to create an incremental
+// sync response for the given device. If the device has a deviceID events
```

Comment:
"... for the given device". But device can be `nil`?

Reply:
"If the device has a deviceID": when does a device not have a deviceID?

Reply:
I think I was getting a bit confused with synapse mainly